### ISSUE TITLE
Python-libnmap Vulnerability

### ISSUE THREAD

**Issue by 0xdade** [natlas/natlas#299]
**Affecting:** 
Both

**Version:** 
b81ae2c

**Describe the bug**
Python-libnmap's developer seems to have just disappeared and left [a vulnerability](https://github.com/advisories/GHSA-9ccv-p7fg-m73x). No response to issues on github. No response via email (as listed in the pypi package information). There's been a vulnerability for nearly a year, and a [pull request](https://github.com/savon-noir/python-libnmap/pull/101) has been available to fix it since February.

If you are in control of all natlas-agents associated with your natlas-server, it is unlikely that this will impact you, provided you have agent authentication turned on. If you do not have agent authentication turned on and anyone on the internet can submit to your `/api/submit`, then there is risk of an XML parsing vulnerability.

**Expected Fix**
It seems like we're going to need to fork python-libnmap into the natlas organization and update it ourselves. Then we'll need to either 1) maintain the package via pip 2) install it via git url e.g. `pipenv install -e git+https://github.com/natlas/python-libnmap.git@0.7.1#egg=python-libnmap` or something similar.

**Comment by 0xdade** [comment]
After some brief work with a clone of this library, it has become obvious that it supports a whole bunch of things that we don't need it to support. I would sooner write a simple nmap-xml module that builds an object representation of a scan result, given an xml file, and then can be serialized back to xml, to json, or whatever else. This could address part of the problem with the current sprawling document schema, as well.

**Comment by 0xdade** [comment]
I have a fixed fork of `python-libnmap` at natlas/python-libnmap. I stripped out everything but the parsing and diffing functionality and I have requested github detach the fork and convert it into a standalone repository, at which point I will change the name to natlas-libnmap and publish it as a package to pypi. I've already tested locally using pip's install-from-git capability, but I'd rather not use install-from-git going forward because we lose the benefit of version identifiers.
